src/ogma/index.ts

```diff
@@ -39,19 +39,26 @@ interface AddItemOptions {
 }
 
 export class LKOgma extends Ogma<LkNodeData, LkEdgeData> {
-  private _reactive: RxViz;
+  private _reactive?: RxViz;
   public LKStyles!: StylesViz;
   public LKCaptions!: CaptionsViz;
-  public LKTransformation: TransformationsViz;
+  public LKTransformation!: TransformationsViz;
   // Trigger an event with node category changes
-  public nodeCategoriesWatcher: NonObjectPropertyWatcher<LkNodeData, LkEdgeData>;
+  public nodeCategoriesWatcher!: NonObjectPropertyWatcher<LkNodeData, LkEdgeData>;
   // Trigger an event with edge type changes
-  public edgeTypeWatcher: NonObjectPropertyWatcher<LkNodeData, LkEdgeData>;
-  public store: OgmaStore;
+  public edgeTypeWatcher!: NonObjectPropertyWatcher<LkNodeData, LkEdgeData>;
+  public store!: OgmaStore;
+  private _configuration: IOgmaConfig;
 
   constructor(configuration: IOgmaConfig) {
     // set Ogma global configuration
     super(configuration);
+    this._configuration = configuration;
+    Object.setPrototypeOf(this, new.target.prototype);
+    this.initOgmaLinkuriusParser();
+  }
+
+  private initOgmaLinkuriusParser(): void {
     this.nodeCategoriesWatcher = this.schema.watchNodeNonObjectProperty({
```

Comment:
There was an old old bug in Ogma v2 where the watchers where not correctly cleared. You should check with the Ogma team how to correctly clear the schema and the watchers.

Reply:
https://github.com/Linkurious/ogma/issues/2827
they are on it. Ogma will make sure that everything is correctly deleted on reset. But we checked the code with @Leo-Nicolle and nothing very annoying should happen!
It is used to update captions so I will test it anyway, to be sure everything is alright!